### PR TITLE
samples: rtc: ds1307: Implementation of RTC API

### PR DESCRIPTION
**Pull request by walidbadar**
This sample application periodically reads date and time data from the DS1307 RTC module, and displays on the console.

Tested with esp32_devkitc_wroom board and DS1307 RTC module.

### CODE REVIEW

**Review by faxe1008 — CHANGES_REQUESTED** [review]
~~There is already a driver for this intree:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers%2Frtc%2Frtc_ds1307.c~~

@walidbadar  Nevermind I misinterpreted the title of the pull request. A general RTC sample would be neat, if there is a board that has an RTC already included. Or maybe just a `build_all` test for them, as some of the drivers are not tested by CI.

File: samples/drivers/rtc/ds1307/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (faxe1008) [inline samples/drivers/rtc/ds1307/LICENSE:1]: A dedicated LICENSE file is not needed, files can just contain a SPDX license header at the top, you can find this in all other intree files. Also locally you can use checkpatch to test your changes before submitting.

File: samples/drivers/rtc/ds1307/src/main.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	/* Check if the RTC is ready */
+	if (!device_is_ready(dev)) {
+		printk("Device is not ready\n");
+		return 0;
+	}
+
+	/* Set the date and time once. */
+	// set_date_time(dev);
```
Comment (faxe1008) [inline samples/drivers/rtc/ds1307/src/main.c:74]: When submitting check that you did not comment out code. 
Also extracting this into a separate function does not bring any benefit in this case imo.

File: samples/drivers/rtc/ds1307/src/main.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+int main(void)
+{
+	int32_t ret = 0;
+
+	/* Get the device structure for the RTC */
+	const struct device *const dev = DEVICE_DT_GET(RTC_NODE);
+
+	/* Check if the RTC is available */
+	if (dev == NULL) {
```
Comment (faxe1008) [inline samples/drivers/rtc/ds1307/src/main.c:62]: Above macro would error during build time if could not get the pointer, so a check here is not needed.

File: samples/drivers/rtc/ds1307/src/main.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		tm.tm_hour, tm.tm_min, tm.tm_sec);
+
+	return ret;
+}
+
+int main(void)
+{
+	int32_t ret = 0;
+
+	/* Get the device structure for the RTC */
```
Comment (faxe1008) [inline samples/drivers/rtc/ds1307/src/main.c:58]: Generally speaking comments are fine for tricky parts but overdoing it produces noise. Getting a device pointer is an operation which is done thousands of times in tree, so comments are not needed.

File: samples/drivers/rtc/ds1307/README.rst [hunk 5]
```diff
@@ -0,0 +1,33 @@
+DS1307: RTC Module
+#######################################################
```
Comment (faxe1008) [inline samples/drivers/rtc/ds1307/README.rst:2]: In RST generally you make the heading specifier just as long as your actual heading.

**Review by faxe1008 — COMMENTED** [review]
In general I think having the RTC drivers be build during CI would be a good thing. If this should be a sample or rather a build all test I'd leave to @bjarki-andreasen :^)

Just left some comments. In general you might want to read the contribution guidelines:
https://docs.zephyrproject.org/latest/contribute/guidelines.html

**Comment by walidbadar** [comment]
@faxe1008  Sorry about the mess, this is my first pull request so bare with me also git does not like windows whitespaces.
Just to clarify it's just implementation of rtc driver not the actual driver.

**Pull request closed without merging**